### PR TITLE
Cisco 9800 WLC Models

### PR DESCRIPTION
**Pull request by arcticash**
Updated the existing C9800-40 with external power supplies as modules (as not part of the chassis)
Adding the other Cisco 9800 WLC Variants
Adding the associated optional C9800 Uplink Modules
Adding the associated external Power Supply Modules

### CODE REVIEW

File: device-types/Cisco/C9800-40-K9.yaml [hunk 1]
```diff
@@ -14,21 +15,24 @@ interfaces:
...
-power-ports:
-  - name: PS1
-    type: iec-60320-c14
-    maximum_draw: 381
-  - name: PS2
-    type: iec-60320-c14
-    maximum_draw: 381
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: PEM 0 Power Supply
```
Comment (danner26) [inline device-types/Cisco/C9800-40-K9.yaml:33]: The names of the PSU module bays should match exactly what either A) the config shows, or B) the documentation shows

The documentation says `PEM 0`, so unless you can get a sample from the configuration lets make them `PEM 0`. If you can get a config output please paste it here so we have it for the future

File: device-types/Cisco/C9800-80-K9.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet0/0/7
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: Modular uplink
+    position: '1'
+  - name: PEM 0 Power Supply
+    label: Bottom Power Supply
+    position: '2'
+  - name: PEM 1 Power Supply
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:47]: The names of the PSU module bays should match exactly what either A) the config shows, or B) the documentation shows

The documentation says `PEM 1`, so unless you can get a sample from the configuration lets make them `PEM 1`. If you can get a config output please paste it here so we have it for the future

File: device-types/Cisco/C9800-80-K9.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - name: TenGigabitEthernet0/0/5
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet0/0/6
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet0/0/7
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: Modular uplink
+    position: '1'
+  - name: PEM 0 Power Supply
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:44]: The names of the PSU module bays should match exactly what either A) the config shows, or B) the documentation shows

The documentation says `PEM 0`, so unless you can get a sample from the configuration lets make them `PEM 0`. If you can get a config output please paste it here so we have it for the future

File: device-types/Cisco/C9800-80-K9.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    label: Bottom Power Supply
+    position: '2'
+  - name: PEM 1 Power Supply
+    label: Top Power Supply
+    position: '3'
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:49]: I think `position` should be `1` for `PEM 1`

Thoughts?

File: device-types/Cisco/C9800-80-K9.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  - name: TenGigabitEthernet0/0/6
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet0/0/7
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: Modular uplink
+    position: '1'
+  - name: PEM 0 Power Supply
+    label: Bottom Power Supply
+    position: '2'
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:46]: I think `position` should be `0` for `PEM 0`

Thoughts?

File: device-types/Cisco/C9800-80-K9.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+airflow: front-to-rear
+is_full_depth: false
+comments: '[Cisco Catalyst 9800-80 Wireless Controller Data Sheet](https://www.cisco.com/c/en/us/products/collateral/wireless/catalyst-9800-series-wireless-controllers/nb-06-cat9800-wirel-cont-data-sheet-ctp-en.html)'
+console-ports:
+  - name: console
+    type: rj-45
+  - name: usb0
+    type: usb-micro-b
+interfaces:
+  - name: Management Port
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:16]: In the configuration, what is the actual ports named? Is it actually called `Management Port`?

File: device-types/Cisco/C9800-80-K9.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    type: 1000base-t
+    mgmt_only: true
+  - name: Redundancy Ethernet Port
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:19]: In the configuration, what is the actual ports named? Is it actually called `Redundancy Ethernet Port`?

File: device-types/Cisco/C9800-80-K9.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    type: 1000base-t
+    mgmt_only: true
+  - name: Redundancy SFP Port
```
Comment (danner26) [inline device-types/Cisco/C9800-80-K9.yaml:22]: In the configuration, what is the actual ports named? Is it actually called `Redundancy SFP Port`?

File: device-types/Cisco/C9800-40-K9.yaml [hunk 9]
```diff
@@ -14,21 +15,24 @@ interfaces:
   - name: Management Port
     type: 1000base-t
     mgmt_only: true
-  - name: Redundancy Port
+  - name: Redundancy Ethernet Port
```
Comment (danner26) [inline device-types/Cisco/C9800-40-K9.yaml:18]: In the configuration, what is the actual ports named? Is it actually called `Redundancy Ethernet Port`?

File: device-types/Cisco/C9800-40-K9.yaml [hunk 10]
```diff
@@ -14,21 +15,24 @@ interfaces:
...
     type: 1000base-t
     mgmt_only: true
+  - name: Redundancy SFP Port
```
Comment (danner26) [inline device-types/Cisco/C9800-40-K9.yaml:21]: In the configuration, what is the actual ports named? Is it actually called `Redundancy SFP Port`?

File: device-types/Cisco/C9800-40-K9.yaml [hunk 11]
```diff
@@ -14,21 +15,24 @@ interfaces:
   - name: Management Port
```
Comment (danner26) [inline device-types/Cisco/C9800-40-K9.yaml:15]: In the configuration, what is the actual ports named? Is it actually called `Management Port`?

File: device-types/Cisco/C9800-L-F.yaml [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+u_height: 1
+is_full_depth: false
+comments: '[Cisco Catalyst 9800-L Wireless Controller Data Sheet](https://www.cisco.com/c/en/us/products/collateral/wireless/catalyst-9800-series-wireless-controllers/datasheet-c78-742434.html)'
+console-ports:
+  - name: Con
+    type: rj-45
+  - name: usb0
+    type: usb-micro-b
+interfaces:
+  - name: Service Port
```
Comment (danner26) [inline device-types/Cisco/C9800-L-F.yaml:17]: In the configuration, what is the actual ports named? Is it actually called `Service Port`?

File: device-types/Cisco/C9800-L-F.yaml [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+    type: 1000base-t
+    mgmt_only: true
+  - name: Redundancy Port
```
Comment (danner26) [inline device-types/Cisco/C9800-L-F.yaml:20]: In the configuration, what is the actual ports named? Is it actually called `Redundancy Port`?

File: device-types/Cisco/C9800-L-F.yaml [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    type: 2.5gbase-t
+  - name: GigabitEthernet2
+    type: 2.5gbase-t
+  - name: GigabitEthernet3
+    type: 2.5gbase-t
+  - name: GigabitEthernet4
+    type: 2.5gbase-t
+power-ports:
+  - name: PS1
+    type: iec-60320-c14
```
Comment (danner26) [inline device-types/Cisco/C9800-L-F.yaml:37]: According to the manufacturer specification, this is a dc-terminal with an external AC to DC adapter. Please change this to `dc-terminal`

File: device-types/Cisco/C9800-L-F.yaml [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    maximum_draw: 110
```
Comment (danner26) [inline device-types/Cisco/C9800-L-F.yaml:38]: According to the manufacturer specification, the maximum draw is `84.5'

> 9800-L-F max measured power = 84.5W (assumes 2pc 2.5W SFP and with 4.5W USB load)

**Review by danner26 — CHANGES_REQUESTED** [review]

File: device-types/Cisco/C9800-80-K9.yaml (hunk 6 above)
Comment (arcticash) [reply device-types/Cisco/C9800-80-K9.yaml:16]: Sorry, don't have access to a config at the moment, I had based the name "Management Port" off another model of 9800 already in the library, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-80-K9.yaml (hunk 7 above)
Comment (arcticash) [reply device-types/Cisco/C9800-80-K9.yaml:19]: Sorry, don't have access to a config at the moment, I had based the name "Management Port" off another model of 9800 already in the library, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-80-K9.yaml (hunk 8 above)
Comment (arcticash) [reply device-types/Cisco/C9800-80-K9.yaml:22]: Sorry, don't have access to a config at the moment, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-40-K9.yaml (hunk 9 above)
Comment (arcticash) [reply device-types/Cisco/C9800-40-K9.yaml:18]: Sorry, don't have access to a config at the moment, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-40-K9.yaml (hunk 10 above)
Comment (arcticash) [reply device-types/Cisco/C9800-40-K9.yaml:21]: Sorry, don't have access to a config at the moment, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-L-F.yaml (hunk 12 above)
Comment (arcticash) [reply device-types/Cisco/C9800-L-F.yaml:17]: Sorry, don't have access to a config at the moment, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-L-F.yaml (hunk 13 above)
Comment (arcticash) [reply device-types/Cisco/C9800-L-F.yaml:20]: Sorry, don't have access to a config at the moment, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-40-K9.yaml (hunk 11 above)
Comment (arcticash) [reply device-types/Cisco/C9800-40-K9.yaml:15]: Sorry, don't have access to a config at the moment, I had based the name "Management Port" off another model of 9800 already in the library, will try and find out the real name of this interface.

File: device-types/Cisco/C9800-L-F.yaml (hunk 12 above)
Comment (danner26) [reply device-types/Cisco/C9800-L-F.yaml:17]: Please let me know when you have found this information

File: device-types/Cisco/C9800-L-F.yaml (hunk 13 above)
Comment (danner26) [reply device-types/Cisco/C9800-L-F.yaml:20]: Please let me know when you have found this information

File: device-types/Cisco/C9800-40-K9.yaml (hunk 11 above)
Comment (danner26) [reply device-types/Cisco/C9800-40-K9.yaml:15]: Please let me know when you have found this information

File: device-types/Cisco/C9800-80-K9.yaml (hunk 6 above)
Comment (danner26) [reply device-types/Cisco/C9800-80-K9.yaml:16]: Please let me know when you have found this information

File: device-types/Cisco/C9800-80-K9.yaml (hunk 7 above)
Comment (danner26) [reply device-types/Cisco/C9800-80-K9.yaml:19]: Please let me know when you have found this information

File: device-types/Cisco/C9800-80-K9.yaml (hunk 8 above)
Comment (danner26) [reply device-types/Cisco/C9800-80-K9.yaml:22]: Please let me know when you have found this information

File: device-types/Cisco/C9800-40-K9.yaml (hunk 9 above)
Comment (danner26) [reply device-types/Cisco/C9800-40-K9.yaml:18]: Please let me know when you have found this information

File: device-types/Cisco/C9800-40-K9.yaml (hunk 10 above)
Comment (danner26) [reply device-types/Cisco/C9800-40-K9.yaml:21]: Please let me know when you have found this information

**Pull request closed without merging**